Index: Makefile
===================================================================
--- Makefile	(revision 49)
+++ Makefile	(working copy)
@@ -1,11 +1,12 @@
 all:
-	gcc -Wall -g segmenter.c -o segmenter -lavformat -lavcodec -lavutil
+	gcc -Wall -g -I/opt/local/include -L/opt/local/lib segmenter.c -o segmenter -lavformat -lavcodec -lavutil
 
 clean:
 	rm segmenter
 
 install: segmenter
-	cp segmenter /usr/local/bin/
+	mkdir -p $(DESTDIR)/opt/local/bin
+	cp segmenter $(DESTDIR)/opt/local/bin
 
 uninstall:
-	rm /usr/local/bin/segmenter
+	rm /opt/local/bin/segmenter
